### PR TITLE
Feature/start end date login 1

### PR DESCRIPTION
**Pull request by borkarsaish65**

### CODE REVIEW

**Review by aks30 — APPROVED** [review]
Approved

File: module/programs/helper.js [hunk 1]
```diff
@@ -73,7 +74,7 @@ module.exports = class ProgramsHelper {
    * @returns {JSON} - create program.
    */
 
-  static create(data) {
+  static create(data, checkDate = false) {
```
Comment (VISHNUDAS-tunerlabs) [inline module/programs/helper.js:78]: @borkarsaish65 Better to update function signature as well

File: module/programs/helper.js [hunk 2]
```diff
@@ -330,7 +350,7 @@ module.exports = class ProgramsHelper {
    * @returns {JSON} - update program.
    */
 
-  static update(programId,data,userId) {
+  static update(programId,data,userId, checkDate = false) {
```
Comment (VISHNUDAS-tunerlabs) [inline module/programs/helper.js:355]: @borkarsaish65 Please update function signature

File: module/solutions/helper.js [hunk 3]
```diff
@@ -398,16 +420,18 @@ module.exports = class SolutionsHelper {
    * @returns {JSON} solution creation data. 
    */
   
-   static update(solutionId, solutionData, userId) {
+   static update(solutionId, solutionData, userId, checkDate = false) {
```
Comment (VISHNUDAS-tunerlabs) [inline module/solutions/helper.js:425]: @borkarsaish65 Please add checkDate to function signature

File: models/programs.js [hunk 4]
```diff
@@ -17,6 +17,16 @@ module.exports = {
         type : String,
         index : true
       },
+      startDate:{
```
Comment (VISHNUDAS-tunerlabs) [inline models/programs.js:20]: @borkarsaish65 Hence we updated model with new keys, It is ideal to update the model in other services as well.

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]
Reviewed on Dec 19 : 10:43

File: module/programs/helper.js (hunk 1 above)
Comment (borkarsaish65) [reply module/programs/helper.js:78]: done

File: module/solutions/helper.js (hunk 3 above)
Comment (borkarsaish65) [reply module/solutions/helper.js:425]: done

File: models/programs.js (hunk 4 above)
Comment (borkarsaish65) [reply models/programs.js:20]: done

File: module/programs/helper.js (hunk 2 above)
Comment (borkarsaish65) [reply module/programs/helper.js:355]: done

**Review by VISHNUDAS-tunerlabs — APPROVED** [review]

**Review by aks30 — APPROVED** [review]
Approved

**Pull request merged**